arrow-inject-compiler-plugin/src/main/kotlin/arrow/inject/compiler/plugin/fir/ProofKey.kt

```diff
@@ -2,4 +2,13 @@ package arrow.inject.compiler.plugin.fir
 
 import org.jetbrains.kotlin.fir.declarations.FirPluginKey
 
+/**
+ * (FIR API review and comments)
+ *
+ * # what's the plugin key for?
+ *
+ * We have used this one across the frontend to set the `origin`
```

Comment:
Keys can be used for passing some information from FIR to IR. We save a plugin key in `IrDeclarationOrigin`, so you can create it in frontend generator and then read it from the IR for this declaration

We're considering different approaches for such data translation (e.g. by introducing declaration attributes for IR declarations), but for now this is the only way except some external service